lib/Ohmbrewer_Temperature_Sensor.cpp

```diff
@@ -149,11 +172,87 @@ bool Ohmbrewer::TemperatureSensor::isOff() const {
 /**
  * Performs the TemperatureSensor's current task. Expect to use this during loop().
  * This function is called by work().
+ *
+ * This analyzes the connected DS18B20 probes and updates temperature.from_c with the Celsius value.
+ *
+ *
  * @returns The time taken to run the method
  */
 int Ohmbrewer::TemperatureSensor::doWork() {
-    // TODO: Implement TemperatureSensor::doWork
-    return -1;
+    //char uid[8];                   //need the unique ID of the probe to search for.
+    uint8_t sensors[80];
+
+    int startTime = millis();        //starting time of reading temperature data
+    uint8_t subzero, cel, celFracBits;        //local vars
+    char msg[100];
+    double tempC;
+    //log("Starting measurement");
+    //Asks all DS18x20 devices to start temperature measurement, takes up to 750ms at max resolution
+    DS18X20_start_meas( DS18X20_POWER_PARASITE, NULL );
+    //If your code has other tasks, you can store the timestamp instead and return when a second has passed.
+    delay(1000);                                        //TODO change / manage this delay
+    uint8_t numSensors = ow_search_sensors(10, sensors);
+    //sprintf(msg, "Found %i sensors", numSensors);
+    //log(msg);
+
+    /*
+     * This section is where we can filter for the desired probe UID and only report that one. for now we will simply
+     * state that there is only one and use sensors[0] as the probe reading.
+     */
+
+    if ( DS18X20_read_meas( &sensors[0], &subzero, &cel, &celFracBits) == DS18X20_OK ) {
+        char sign = (subzero) ? '-' : '+';
+        int frac = celFracBits*DS18X20_FRACCONV;
+        tempC = (double)cel;
+        if (sign == '-'){
+            tempC = tempC * -1;
+        }
+        tempC = tempC + (.0001*(double)frac);
+        getTemp()->fromC(tempC);
+    }
+
+    ///////////////Code below replaces above block when infrastructure is ready/////////////////
+/*
+    // filters for a desired probe.
+    for (uint8_t i=0; i<numSensors; i++){
+        //current probe ID
+        char probeId[8] = {
+				sensors[(i*OW_ROMCODE_SIZE)+0],
+				sensors[(i*OW_ROMCODE_SIZE)+1],
+				sensors[(i*OW_ROMCODE_SIZE)+2],
+				sensors[(i*OW_ROMCODE_SIZE)+3],
+				sensors[(i*OW_ROMCODE_SIZE)+4],
+				sensors[(i*OW_ROMCODE_SIZE)+5],
+				sensors[(i*OW_ROMCODE_SIZE)+6],
+				sensors[(i*OW_ROMCODE_SIZE)+7]};
+
```

Comment:
i dont get what you mean, this code block is the functional code for this method. and we have to pull it everytime we read in order to establish that our probe is still connected. 
I am leaving this comented out here so that once we add the set up capibilities elsewhere in the program we can easily change the fnutionality.
Unless you mean make a duplicate function that we can change the call to when we do the above mentioned.?